specification/cognitiveservices/ContentSafetyInternal/models.tsp

```diff
@@ -0,0 +1,232 @@
+import "@typespec/http";
+import "@typespec/rest";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+
+
+namespace ContentSafetyInternal;
+
+@doc("Whitelist features.")
+@resource("whitelist/features")
+model Feature {
+  #suppress "@azure-tools/typespec-azure-core/key-visibility-required" "Properties in response-only schemas should not be marked readOnly"
+  @key("featueName")
+  @doc("Feature Name.")
+  featureName: string;
+
+  @doc("The properties of the feature.")
+  properties: object;
+}
+
+@doc("Response of listing whitelist features.")
+model FeatureList {
+  @doc("List of whitelist features.")
+  value: Feature[];
+}
+
+// Models ////////////////////
+@doc("Time granularity.")
+enum Granularity {
+  Hourly,
+  Daily
+}
+
+@doc("Modality type.")
+enum ContentModality {
+  Text,
+  Image,
+  @added(ContentSafetyInternal.Versions.v2023_05_30_Preview)
+  @removed(ContentSafetyInternal.Versions.v2023_10_01)
+  ImageWithText
+}
+
+@doc("Content safety Operation.")
+enum ContentSafetyOperation {
+  AnalyzeText,
+  AnalyzeImage,
+  @added(ContentSafetyInternal.Versions.v2023_05_30_Preview)
+  @removed(ContentSafetyInternal.Versions.v2023_10_01)
+  AnalyzeImageWithText
+}
+
+@doc("Content safety API version.")
+enum ContentSafetyVersion {
```

Comment:
As SDK owner, we advise to make enum name for each value like: 
```
@doc("Content safety API version.")
enum ContentSafetyVersion {
  v2023_04_30_Preview: "2023-04-30-preview",
  @added(ContentSafetyInternal.Versions.v2023_05_30_Preview)
  @removed(ContentSafetyInternal.Versions.v2023_10_01)
  v2023_05_30_Preview: "2023-05-30-preview",
  @added(ContentSafetyInternal.Versions.v2023_10_01)
  v2023_10_01: "2023-10-01"
}
```

Reply:
Thanks for review. This a PR for internal review. I will abondon the PR to reduce your work.